packages/beacon-node/src/chain/observer.ts

```diff
@@ -73,19 +93,35 @@ export abstract class ChainObserver extends ObserverHandlers {
 }
 
 /**
- * The queue observer push the events to a queue and then process those one by one from the queue
+ * A `ChainObserver` variant that processes events via an internal queue.
+ *
+ * Instead of calling the event handlers immediately, it enqueues them
+ * in a `JobItemQueue`. Items in this queue are processed one by one
+ * (or in batches, if you adapt `JobItemQueue`). This is useful for ensuring
+ * that event handling is sequential and does not block other operations.
  */
 export abstract class QueueObserver extends ChainObserver {
   protected jobQueue: JobItemQueue<[HandlerNames, unknown[]], void>;
 
+  /**
+   * Constructs a `QueueObserver` with a maximum queue length and
+   * an optional AbortSignal to terminate processing.
+   */
   constructor({maxQueueLength, signal, logger}: {maxQueueLength: number; signal: AbortSignal; logger: Logger}) {
     super({logger});
 
     this.jobQueue = new JobItemQueue(
       async (handler, args) => {
         const eventHandler = this[handler];
-        // biome-ignore lint/suspicious/noExplicitAny: Can not use `unknown` type here because of union
-        if (eventHandler) await eventHandler(...(args as any[]));
+        if (!eventHandler) return;
+
+        try {
+          // biome-ignore lint/suspicious/noExplicitAny: Can not use `unknown` type here because of union
+          await eventHandler(...(args as any[]));
+        } catch (err) {
+          // We had to catch and log because we want to keep processing the rest of the events
+          this.logger.error("Queue event caused error", {}, err as Error);
+        }
```

Comment:
What happens when events are dependent on prior events (like the block/state timing issue from the comment we moved?).  Seems like this may cause undesired side effects (will not restart properly in the case of that comment)